### PR TITLE
Add split path in the cursor error message

### PR DESCRIPTION
**Pull request by Yaliang**

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java [hunk 1]
```diff
@@ -92,6 +92,8 @@ public ThriftHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, ThriftFie
 
         return Optional.of(new ThriftHiveRecordCursor<>(
                 genericRecordReader(recordReader),
+                path,
+                start,
```
Comment (dabaitu) [inline presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java]: Just pure personal opinion: seems these two params are passed in just for a more detailed exception message. I personally prefer not to mix in anything if they aren't needed logically.
What do you think of just try-catch in CursorProvider and throw the detailed exception here?

Also I think if we just go with String.format we avoid having to add a line for import. :)

Comment (Yaliang) [inline presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java]: Here is just creating the record cursor and not the place to actually read the records. If we want to display the detailed split path and range, we need to pass these two parameters into the cursor. And add them into the error message in the "advanceNextPosition" method, which is the public method for interface `com.facebook.presto.spi.RecordCursor` and called by the `getNextPage` in the `com.facebook.presto.spi.RecordPageSource`

Comment (dabaitu) [inline presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java]: Oh I see. I didn't expand, thought the exception was in the constructor. I see it's in advanceNextPosition.

Comment (Yaliang) [inline presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java]: Got it. Merging it.

**Pull request merged**